starter-code/src/clue.js

```diff
@@ -1,94 +1,201 @@
+ age:          36,
+ image:        'https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg',
+ occupation:   'Socialité'
+ }
+ 
+ let mrMustard = {
+ first_name:   'Jack',
+ last_name:    'Mustard',
+ color:        'yellow',
+ description:  'He is a former football player who tries to get by on his former glory',
+ age:          62,
+ image:        'https://metrouk2.files.wordpress.com/2016/07/colonel-mustard.jpg',
+ occupation:   'Retired Football player'
+ }
+ 
+ var charactersArray = [];
+ 
+charactersArray.push(drOrchid,mrsPeacock,mrMustard,missScarlet, mrGreen, profPlum);
+
+//console.log(charactersArray);
 
 // Weapons
 
-name: rope        ---   weight: 10
-name: knife       ---   weight: 8
-name: candlestick ---   weight: 2
-name: dumbbell    ---   weight: 30
-name: poison      ---   weight: 2
-name: axe         ---   weight: 15
-name: bat         ---   weight: 13
-name: trophy      ---   weight: 25
-name: pistol      ---   weight: 20
-
-// Rooms
-name: Dinning Room
-name: Conservatory
-name: Kitchen
-name: Study
-name: Library
-name: Billiard Room
-name: Lounge
-name: Ballroom
-name: Hall
-name: Spa
-name: Living Room
-name: Observatory
-name: Theater
-name: Guest House
-name: Patio
-*/
-
-// Characters Collection
-var charactersArray = [];
+let rope = {
+  name: 'rope',
+  weight: 10,
+}
 
-// Rooms' Collection
-var roomsArray = [];
+let knife = {
+  name: 'knife',
+  weight: 8
+}
+
+let candleStick = {
+  name: 'candlestick',
+  weight: 2
+}
+
+let dumbbell = {
+  name: 'dumbbell',
+  weight: 30
+}
+let poison = {
+  name: 'poison',
+  weight: 2
+}
+
+let axe = {
+  name: 'axe',
+  weight: 15
+}
+
+let bat = {
+  name: 'bat',
+  weight: 13
+}
+
+let trophy = {
+  name: 'trophy',
+  weight: 25
+}
+
+let pistol = {
+  name: 'pistol',
+  weight: 20
+}
 
 // Weapons Collection
 var weaponsArray = [];
+
+weaponsArray.push(rope, knife, candleStick, dumbbell, poison, axe, bat, trophy, pistol)
+
+//console.log(weaponsArray)
+
+let diningRoom = {
+  name: "Dinning Room"
+}
+let conservatory = {
+  name: "Conservatory"
+}
+let kitchen = {
+  name: "Kitchen"
+}
+let study = {
+  name: "Study"
+}
+let library = {
+  name: "Library"
+}
+let billiardRoom = {
+  name: "Billiard Room"
+}
+let lounge = {
+  name: "Lounge"
+}
+let ballroom = {
+  name: "Ballroom"
+}
+let hall = {
+  name: "Hall"
+}
+let spa = {
+  name: "Spa"
+}
+let livingRoom = {
+  name: "Living Room"
+}
+let observatory = {
+  name: "Observatory"
+}
+let theater = {
+  name: "Theater"
+}
+let guestHouse = {
+  name: "Guest House"
+}
+let patio = {
+  name: "Patio"
+}
+
+// Rooms' Collection
+let roomsArray = [];
+
+roomsArray.push(diningRoom, conservatory, kitchen, study, library, billiardRoom, 
+lounge, ballroom, hall, spa, livingRoom, observatory, theater, guestHouse, patio);
+
+function randomSelector(array){
+  let random = 0;
+  
+    random = Math.floor(Math.random() * array.length);
+  return array[random];
+}
+
+
+
+function pickMistery(){
```

Comment:
You almost had it all Perfect  😁... Only thing missing is the very, end of it which would be written like this:

```js
function pickMistery() {
  var killer = randomSelector(charactersArray);
  var weapon = randomSelector(weaponsArray);
  var room = randomSelector(roomsArray);

  var resultArray = [];
  resultArray.push(killer, weapon, room);
  return resultArray;
}

function revealMistery(mistery) {
  return mistery[0].first_name + ' ' + mistery[0].last_name + ' killed Mr.Boddy using the ' + mistery[1].name + ' in the ' + mistery[2].name + '!!!!';
}

var misteryEnvelope = pickMistery();
```